### PR TITLE
Feature/596 html modal

### LINKED ISSUE #596: Feedback/Modal
### User story

Create HTML & CSS version of the modal. https://www.figma.com/file/JpoY3waVoQGlLQzQXTL9nn/Design-System---Gemeente-Den-Haag?node-id=0%3A1

### PR DESCRIPTION
**Pull request by paulAcato**
Modal HTML, CSS and JS parts.

Screenshot 2021-12-09 at 15 03 08

closes #596

### CODE REVIEW

**Review by bddjong — CHANGES_REQUESTED** [review]

File: components/Modal/src/modal.css [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+  color: var(--denhaag-modal-icon-color);
+  cursor: pointer;
+  line-height: 1;
+  position: absolute;
+  top: 0;
+  right: 0;
+  padding-inline-start: var(--denhaag-modal-icon-padding-inline, var(--denhaag-modal-padding));
+  padding-inline-end: var(--denhaag-modal-icon-padding-inline, var(--denhaag-modal-padding));
+  padding-block-start: var(--denhaag-modal-icon-padding-block, var(--denhaag-modal-padding));
+  padding-block-end: var(--denhaag-modal-icon-padding-block, var(--denhaag-modal-padding));
```
[the remark this answers is not in the document]
Comment (paulAcato) [reply components/Modal/src/modal.css:64]: @bddjong , I used the button as example,

When I check the button component I saw the following;
`padding-block-end: var(--denhaag-button-medium-size-padding-block, var(--denhaag-button-padding-block));
  padding-block-start: var(--denhaag-button-medium-size-padding-block, var(--denhaag-button-padding-block));
  padding-inline-end: var(--denhaag-button-medium-size-padding-inline, var(--denhaag-button-padding-inline));
  padding-inline-start: var(--denhaag-button-medium-size-padding-inline, var(--denhaag-button-padding-inline));`
  
 Therefor I applied the same logic with the `var(--denhaag-modal-padding);` as fallback. But i'll change it, no problem

File: components/Modal/src/modal.css [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+}
+
+.denhaag-modal__button svg {
+  font-size: var(--denhaag-modal-icon-size, var(--denhaag-modal-padding));
```
[the remark this answers is not in the document]
Comment (paulAcato) [reply components/Modal/src/modal.css:68]: I missed this, should be inherit as fallback.

**Comment by bddjong** [comment]
Is it correct that the opening / closing javascript doesn't work in storybook?

**Comment by paulAcato** [comment]
> Is it correct that the opening / closing javascript doesn't work in storybook?

@bddjong , they should work. At my place they're working? See the attached video

https://user-images.githubusercontent.com/94894596/147116513-1059cd9d-46fc-426b-9eb7-27dae474ade5.mp4

**Comment by bddjong** [comment]
> > Is it correct that the opening / closing javascript doesn't work in storybook?
> 
> @bddjong , they should work. At my place they're working? See the attached video

I tested the storybook stories on both Chrome and Firefox. In the case of Firefox, I was unable to even press that buttons (@mjcarsjens can you test this branch to confirm that?). In Chrome, pressing the buttons didn't do anything.

**Comment by bddjong** [comment]
Just tested in Safari. Safari works fine.

**Comment by paulAcato** [comment]
> Just tested in Safari. Safari works fine the first time, but not after switching between Canvas and Docs in storybook.

I just edited the pointer-event, where FireFox broke on.

**Review by bddjong — APPROVED** [review]

**Review by bddjong — CHANGES_REQUESTED** [review]
Only noticed one more issue, looks good otherwise

File: components/Modal/src/index.tsx [hunk 3]
```diff
@@ -0,0 +1,3 @@
+import './modal.css';
+import './stories.css';
+import './stories.js';
```
[the remark this answers is not in the document]
Comment (robbieatacato) [reply components/Modal/src/index.tsx]: fixed it, see commits

**Review by bddjong — APPROVED** [review]

**Pull request merged**